pkgs/development/libraries/qgnomeplatform/hardcode-gsettings.patch

```diff
@@ -0,0 +1,33 @@
+diff --git a/common/gnomehintssettings.cpp b/common/gnomehintssettings.cpp
+index 4be842d..a5581ec 100644
+--- a/common/gnomehintssettings.cpp
++++ b/common/gnomehintssettings.cpp
+@@ -80,8 +80,11 @@ void gtkMessageHandler(const gchar *log_domain,
+ GnomeHintsSettings::GnomeHintsSettings()
+     : QObject(0)
+     , m_usePortal(checkUsePortalSupport())
+-    , m_gnomeDesktopSettings(g_settings_new("org.gnome.desktop.wm.preferences"))
+-    , m_settings(g_settings_new("org.gnome.desktop.interface"))
++    , m_schema_source(g_settings_schema_source_new_from_directory("@gds_gsettings_path@", g_settings_schema_source_get_default(), true, NULL))
++    , m_gnomeDesktopSchema(g_settings_schema_source_lookup(m_schema_source, "org.gnome.desktop.wm.preferences", false))
++    , m_gnomeDesktopSettings(g_settings_new_full(m_gnomeDesktopSchema, NULL, NULL))
++    , m_settingsSchema(g_settings_schema_source_lookup(m_schema_source, "org.gnome.desktop.interface", false))
++    , m_settings(g_settings_new_full(m_settingsSchema, NULL, NULL))
```

Comment:
The issue with this is that it does not free the schema source and schemas. That is too advanced for initializer lists so we need to move all of these into the body of the constructor (and then use [`g_autoptr`](https://developer.gnome.org/glib/stable/glib-Miscellaneous-Macros.html#g-autoptr) on those). Fortunately, we are only dealing with object’s own properties so it should be fine.

Reply:
Also, it looks like the original code is not correctly disposing of [`GSettings`](https://developer.gnome.org/gio/stable/GSettings.html) instances either. They should call [`g_object_unref`](https://developer.gnome.org/gobject/stable/gobject-The-Base-Object-Type.html#g-object-unref) in the [destructor](https://github.com/FedoraQt/QGnomePlatform/blob/a4a5788e754c0d55980454a60a04963a2ca31aa6/common/gnomehintssettings.cpp#L148).

Reply:
Opened https://github.com/FedoraQt/QGnomePlatform/pull/57